src/com/sahokia/kos/handmadecollections/MyArrayList.java

```diff
@@ -0,0 +1,48 @@
+package com.sahokia.kos.handmadecollections;
+
+public class MyArrayList<T> {
+    private Object[] array;
+    private int size;
+    private static final int DEFAULT_CAPACITY = 10;
+
+    public MyArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    public void put(T element) {
+        if (size == array.length) {
+            resize(array.length * 2);
+        }
+        array[size] = element;
+        size++;
+    }
+
+    public T get(int index) {
+        if (index >= 0 && index < size) {
+            return (T) array[index];
+        } else {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+    }
+
+    public void delete(int index) {
+        if (index >= 0 && index < size) {
+            for (int i = index; i < size - 1; i++) {
+                array[i] = array[i + 1];
+            }
+            array[--size] = null;
+            if (size <= array.length / 4 && array.length / 2 >= DEFAULT_CAPACITY) {
```

Comment:
Why we divide by 4?)

Reply:
And after divide by 2?

Reply:
changed this method to be easier to work: if (size < array.length / 2) resize, also we have additional check not to become smaller than initial capacity.

Reply:
Create, please, separate fields for this 'magic' numbers